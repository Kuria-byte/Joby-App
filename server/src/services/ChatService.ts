import { Message, IMessage } from '../models/Message';
import { Match } from '../models/Match';

export class ChatService {
  async saveMessage(matchId: string, senderId: string, content: string): Promise<IMessage> {
    try {
      // Update the match's lastMessageAt
      await Match.findByIdAndUpdate(matchId, {
        lastMessageAt: new Date(),
        status: 'active' // Ensure match is active when messages are exchanged
      });

      // Create and save the new message
      const message = await Message.create({
        matchId,
        senderId,
        content,
        type: 'text'
      });

      return message;
    } catch (error) {
      console.error('Error in saveMessage:', error);
      throw error;
    }
  }

  async getMessageHistory(matchId: string, limit: number = 50): Promise<IMessage[]> {
    try {
      return await Message.find({ matchId })
        .sort({ createdAt: -1 })
        .limit(limit)
        .lean();
    } catch (error) {
      console.error('Error in getMessageHistory:', error);
      throw error;
    }
  }

  async markMessagesAsRead(matchId: string, userId: string): Promise<void> {
    try {
      await Message.updateMany(
        {
          matchId,
          senderId: { $ne: userId },
          readAt: null
        },
        {
          readAt: new Date()
        }
      );
    } catch (error) {
      console.error('Error in markMessagesAsRead:', error);
      throw error;
    }
  }
}
