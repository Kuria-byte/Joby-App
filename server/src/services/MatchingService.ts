import { Match, IMatch } from '../models/Match';

export class MatchingService {
  async checkForMatch(userId: string, jobId: string, employerId: string, jobTitle: string, companyName: string): Promise<IMatch | null> {
    try {
      // Check if employer has already shown interest
      const existingMatch = await Match.findOne({ jobId, userId });
      
      if (existingMatch) {
        // Update match status to active if it was pending
        if (existingMatch.status === 'pending') {
          existingMatch.status = 'active';
          await existingMatch.save();
        }
        return existingMatch;
      }

      // Create a new pending match
      const newMatch = await Match.create({
        jobId,
        userId,
        employerId,
        jobTitle,
        companyName,
        status: 'pending'
      });

      return null; // No mutual match yet
    } catch (error) {
      console.error('Error in checkForMatch:', error);
      throw error;
    }
  }

  async getUserMatches(userId: string): Promise<IMatch[]> {
    try {
      return await Match.find({
        userId,
        status: { $in: ['active', 'pending'] }
      }).sort({ lastMessageAt: -1, matchedAt: -1 });
    } catch (error) {
      console.error('Error in getUserMatches:', error);
      throw error;
    }
  }

  async updateMatchStatus(matchId: string, status: IMatch['status']): Promise<IMatch | null> {
    try {
      return await Match.findByIdAndUpdate(
        matchId,
        { status },
        { new: true }
      );
    } catch (error) {
      console.error('Error in updateMatchStatus:', error);
      throw error;
    }
  }
}
