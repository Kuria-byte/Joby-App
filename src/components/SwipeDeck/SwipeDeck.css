.swipe-deck {
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
  overflow: hidden;
}

.cards-container {
  position: relative;
  width: 100%;
  max-width: 480px;
  height: calc(100vh - 200px);
  display: flex;
  justify-content: center;
  align-items: center;
  perspective: 1000px;
}

.card-wrapper {
  position: absolute;
  width: 100%;
  transition: transform 0.3s ease, opacity 0.3s ease;
  cursor: grab;
  will-change: transform, opacity;
}

.card-wrapper:active {
  cursor: grabbing;
}

/* Prevent text selection during swipe */
.card-wrapper * {
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.swipe-buttons {
  position: relative;
  display: flex;
  justify-content: center;
  gap: 40px;
  margin-top: 20px;
  z-index: 1000;
}

.swipe-button {
  width: 64px;
  height: 64px;
  border-radius: 50%;
  border: none;
  font-size: 24px;
  cursor: pointer;
  transition: all 0.2s ease;
  display: flex;
  align-items: center;
  justify-content: center;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);
}

.swipe-button:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.swipe-button.reject {
  background-color: #ff4b4b;
  color: white;
}

.swipe-button.accept {
  background-color: #4CAF50;
  color: white;
}

.swipe-button:hover:not(:disabled) {
  transform: scale(1.1);
}

.swipe-button:active:not(:disabled) {
  transform: scale(0.95);
}

/* Swipe indicators */
.card-wrapper::before {
  content: '';
  position: absolute;
  top: 20px;
  width: 100px;
  height: 100px;
  background-size: contain;
  background-repeat: no-repeat;
  opacity: 0;
  transition: opacity 0.3s ease;
  z-index: 1000;
}

.card-wrapper[style*="translateX("] {
  transition: transform 0.3s ease, opacity 0.3s ease, rotate 0.3s ease;
}

.processing-overlay {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(255, 255, 255, 0.8);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
}

.processing-spinner {
  width: 40px;
  height: 40px;
  border: 4px solid #f3f3f3;
  border-top: 4px solid #3498db;
  border-radius: 50%;
  animation: spin 1s linear infinite;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

/* Media Queries */
@media (max-width: 480px) {
  .swipe-deck {
    padding: 10px;
  }

  .cards-container {
    height: calc(100vh - 160px);
  }

  .swipe-buttons {
    gap: 20px;
  }

  .swipe-button {
    width: 56px;
    height: 56px;
    font-size: 20px;
  }
}

/* Add smooth transitions for card stack effect */
@keyframes slideIn {
  from {
    transform: translateX(100%);
    opacity: 0;
  }
  to {
    transform: translateX(0);
    opacity: 1;
  }
}

.card-wrapper:not(:first-child) {
  pointer-events: none;
}

.swipe-deck-container {
  position: relative;
  width: 100%;
  max-width: 600px;
  height: 500px;
  margin: 0 auto;
  perspective: 1000px;
  touch-action: none;
}

.card-container {
  position: absolute;
  width: 100%;
  height: 100%;
  cursor: grab;
  will-change: transform;
  transform-origin: center center;
}

.card-container:active {
  cursor: grabbing;
}

.next-card-preview {
  position: absolute;
  width: 100%;
  height: 100%;
  transform: scale(0.95) translateY(10px);
  opacity: 0.5;
  pointer-events: none;
  z-index: -1;
}

.swipe-deck-empty {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  font-size: 1.2rem;
  color: #666;
  text-align: center;
  padding: 2rem;
  background: #f5f5f5;
  border-radius: 12px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
}

/* Add smooth transition for transform changes */
.card-container {
  transition: transform 0.2s ease-out;
}

/* Optional: Add a subtle hover effect */
.card-container:hover {
  transform: translateY(-2px);
  box-shadow: 0 4px 15px rgba(0, 0, 0, 0.1);
}
