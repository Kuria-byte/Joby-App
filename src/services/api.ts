const API_BASE_URL = process.env.REACT_APP_API_BASE_URL || 'http://localhost:3001';

export interface Job {
  id: string;
  title: string;
  company: string;
  location: string;
  description: string;
}

export const api = {
  async getJobs(): Promise<Job[]> {
    const response = await fetch(`${API_BASE_URL}/jobs`);
    if (!response.ok) {
      throw new Error('Failed to fetch jobs');
    }
    return response.json();
  },

  async likeJob(jobId: string): Promise<void> {
    const response = await fetch(`${API_BASE_URL}/jobs/${jobId}/like`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    if (!response.ok) {
      throw new Error('Failed to like job');
    }
  },

  async passJob(jobId: string): Promise<void> {
    const response = await fetch(`${API_BASE_URL}/jobs/${jobId}/pass`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    if (!response.ok) {
      throw new Error('Failed to pass job');
    }
  },
};
