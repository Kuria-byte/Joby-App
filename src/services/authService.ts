import axios from 'axios';

export interface LoginCredentials {
  email: string;
  password: string;
}

export interface SignupCredentials extends LoginCredentials {
  name: string;
  confirmPassword: string;
}

export interface AuthResponse {
  token: string;
  user: {
    id: string;
    name: string;
    email: string;
  };
}

class AuthService {
  private baseURL = '/api/auth';
  private tokenKey = 'joby_auth_token';

  async login(credentials: LoginCredentials): Promise<AuthResponse> {
    try {
      const response = await axios.post(`${this.baseURL}/login`, credentials);
      this.setToken(response.data.token);
      return response.data;
    } catch (error) {
      if (axios.isAxiosError(error)) {
        if (error.response?.status === 401) {
          throw new Error('Invalid email or password');
        }
        if (error.response?.status === 429) {
          throw new Error('Too many login attempts. Please try again later');
        }
      }
      throw new Error('Login failed. Please try again');
    }
  }

  async signup(credentials: SignupCredentials): Promise<AuthResponse> {
    if (credentials.password !== credentials.confirmPassword) {
      throw new Error('Passwords do not match');
    }

    try {
      const response = await axios.post(`${this.baseURL}/signup`, credentials);
      this.setToken(response.data.token);
      return response.data;
    } catch (error) {
      if (axios.isAxiosError(error)) {
        if (error.response?.status === 409) {
          throw new Error('Email already exists');
        }
        if (error.response?.status === 400) {
          throw new Error('Invalid signup information');
        }
      }
      throw new Error('Signup failed. Please try again');
    }
  }

  async logout(): Promise<void> {
    try {
      await axios.post(`${this.baseURL}/logout`);
    } finally {
      this.removeToken();
    }
  }

  async validateToken(): Promise<boolean> {
    const token = this.getToken();
    if (!token) return false;

    try {
      await axios.get(`${this.baseURL}/validate`, {
        headers: { Authorization: `Bearer ${token}` }
      });
      return true;
    } catch {
      this.removeToken();
      return false;
    }
  }

  getToken(): string | null {
    return localStorage.getItem(this.tokenKey);
  }

  private setToken(token: string): void {
    localStorage.setItem(this.tokenKey, token);
  }

  private removeToken(): void {
    localStorage.removeItem(this.tokenKey);
  }
}

export const authService = new AuthService();
