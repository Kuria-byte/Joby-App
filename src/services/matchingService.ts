import { Job } from './api';

export interface Match {
  id: string;
  jobId: string;
  userId: string;
  employerId: string;
  jobTitle: string;
  companyName: string;
  matchedAt: string;
  status: 'pending' | 'active' | 'closed';
}

export interface MatchResponse {
  isMatch: boolean;
  match?: Match;
}

class MatchingService {
  private socket: WebSocket;

  constructor() {
    this.socket = new WebSocket('ws://localhost:3000/matches');
    this.setupSocketListeners();
  }

  private setupSocketListeners() {
    this.socket.onmessage = (event) => {
      const data = JSON.parse(event.data);
      if (data.type === 'match') {
        this.handleNewMatch(data.match);
      }
    };
  }

  private handleNewMatch(match: Match) {
    // Dispatch a custom event that components can listen to
    const matchEvent = new CustomEvent('newMatch', { detail: match });
    window.dispatchEvent(matchEvent);
  }

  async checkForMatch(userId: string, jobId: string): Promise<MatchResponse> {
    try {
      const response = await fetch('http://localhost:3000/api/matches/check', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ userId, jobId }),
      });
      
      if (!response.ok) {
        throw new Error('Failed to check for match');
      }

      return await response.json();
    } catch (error) {
      console.error('Error checking for match:', error);
      throw error;
    }
  }

  async getMatches(userId: string): Promise<Match[]> {
    try {
      const response = await fetch(`http://localhost:3000/api/matches/user/${userId}`);
      if (!response.ok) {
        throw new Error('Failed to fetch matches');
      }
      return await response.json();
    } catch (error) {
      console.error('Error fetching matches:', error);
      throw error;
    }
  }
}

export const matchingService = new MatchingService();
