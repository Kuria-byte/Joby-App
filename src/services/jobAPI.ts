import axios from 'axios';
import { Job } from './api';
import { matchingService, MatchResponse } from './matchingService';

const API_BASE_URL = 'http://localhost:3000/api'; // Change this to your actual API base URL

export interface JobInterestData {
  jobId: string;
  userId: string;
  action: 'interested' | 'dismissed';
  timestamp: string;
}

export const jobAPI = {
  // Record user's interest in a job
  recordJobInterest: async (jobId: string, userId: string): Promise<MatchResponse> => {
    try {
      const data: JobInterestData = {
        jobId,
        userId,
        action: 'interested',
        timestamp: new Date().toISOString(),
      };
      
      await axios.post(`${API_BASE_URL}/jobs/interest`, data);
      
      // Check for a match after recording interest
      return await matchingService.checkForMatch(userId, jobId);
    } catch (error) {
      console.error('Error recording job interest:', error);
      throw error;
    }
  },

  // Record job dismissal
  dismissJob: async (jobId: string, userId: string): Promise<void> => {
    try {
      const data: JobInterestData = {
        jobId,
        userId,
        action: 'dismissed',
        timestamp: new Date().toISOString(),
      };
      
      await axios.post(`${API_BASE_URL}/jobs/dismiss`, data);
    } catch (error) {
      console.error('Error dismissing job:', error);
      throw error;
    }
  },

  // Get job recommendations
  getJobRecommendations: async (userId: string, limit: number = 10): Promise<Job[]> => {
    try {
      const response = await axios.get<Job[]>(`${API_BASE_URL}/jobs/recommendations`, {
        params: { userId, limit }
      });
      return response.data;
    } catch (error) {
      console.error('Error fetching job recommendations:', error);
      throw error;
    }
  }
};
