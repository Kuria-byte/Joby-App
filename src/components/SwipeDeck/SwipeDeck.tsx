import React, { useState, useRef, useCallback } from 'react';
import { useSwipeable, SwipeEventData } from 'react-swipeable';
import { motion, AnimatePresence } from 'framer-motion';
import JobCard from '../JobCard/JobCard';
import { Job } from '../../services/api';
import { jobAPI } from '../../services/jobAPI';
import { logError, showErrorToast } from '../../utils/errorUtils';
import './SwipeDeck.css';

interface SwipeDeckProps {
  jobs: Job[];
  userId: string;
  onStackEmpty: () => void;
  onError?: (error: Error) => void;
}

const SWIPE_THRESHOLD = 100; // pixels to trigger swipe
const ROTATION_ANGLE = 12; // maximum rotation angle in degrees

const SwipeDeck: React.FC<SwipeDeckProps> = ({ jobs, userId, onStackEmpty, onError }) => {
  const [currentIndex, setCurrentIndex] = useState(0);
  const [swipeDistance, setSwipeDistance] = useState(0);
  const [isProcessing, setIsProcessing] = useState(false);
  const [exitX, setExitX] = useState(0);

  const handleSwipeAction = useCallback(async (direction: 'left' | 'right') => {
    if (isProcessing) return;

    const currentJob = jobs[currentIndex];
    setIsProcessing(true);
    setExitX(direction === 'right' ? 1000 : -1000);

    try {
      if (direction === 'right') {
        await jobAPI.recordJobInterest(currentJob.id, userId);
      } else {
        await jobAPI.dismissJob(currentJob.id, userId);
      }

      setCurrentIndex(prev => {
        const newIndex = prev + 1;
        if (newIndex >= jobs.length) {
          onStackEmpty();
        }
        return newIndex;
      });
    } catch (error) {
      logError(error as Error, 'SwipeDeck', userId, {
        jobId: currentJob.id,
        action: direction
      });
      showErrorToast(error as Error);
      onError?.(error as Error);
    } finally {
      setIsProcessing(false);
      setSwipeDistance(0);
      setExitX(0);
    }
  }, [currentIndex, jobs, userId, isProcessing, onStackEmpty, onError]);

  const handlers = useSwipeable({
    onSwiping: (eventData: SwipeEventData) => {
      if (!isProcessing) {
        setSwipeDistance(eventData.deltaX);
      }
    },
    onSwipedLeft: () => {
      if (Math.abs(swipeDistance) > SWIPE_THRESHOLD) {
        handleSwipeAction('left');
      } else {
        setSwipeDistance(0);
      }
    },
    onSwipedRight: () => {
      if (Math.abs(swipeDistance) > SWIPE_THRESHOLD) {
        handleSwipeAction('right');
      } else {
        setSwipeDistance(0);
      }
    },
    trackMouse: true,
    preventScrollOnSwipe: true,
  });

  if (currentIndex >= jobs.length) {
    return <div className="swipe-deck-empty">No more jobs to show</div>;
  }

  const currentJob = jobs[currentIndex];
  const rotation = (swipeDistance / SWIPE_THRESHOLD) * ROTATION_ANGLE;
  const opacity = Math.max(0, 1 - Math.abs(swipeDistance) / (SWIPE_THRESHOLD * 2));

  return (
    <div className="swipe-deck-container" {...handlers}>
      <AnimatePresence>
        <motion.div
          key={currentJob.id}
          className="card-container"
          initial={{ x: 0, opacity: 1 }}
          animate={{
            x: swipeDistance,
            rotate: rotation,
            opacity: opacity
          }}
          exit={{
            x: exitX,
            opacity: 0,
            transition: { duration: 0.2 }
          }}
          transition={{
            type: "spring",
            damping: 15,
            stiffness: 150
          }}
        >
          <JobCard job={currentJob} />
        </motion.div>
      </AnimatePresence>
      {currentIndex < jobs.length - 1 && (
        <div className="next-card-preview">
          <JobCard job={jobs[currentIndex + 1]} />
        </div>
      )}
    </div>
  );
};

export default SwipeDeck;
