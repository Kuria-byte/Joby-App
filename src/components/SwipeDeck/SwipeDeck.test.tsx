import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SwipeDeck from './SwipeDeck';
import { jobAPI } from '../../services/jobAPI';

// Mock the jobAPI
jest.mock('../../services/jobAPI', () => ({
  jobAPI: {
    recordJobInterest: jest.fn(),
    dismissJob: jest.fn(),
  },
}));

const mockJobs = [
  {
    id: '1',
    title: 'Frontend Developer',
    company: 'Tech Co',
    location: 'New York',
    description: 'Frontend role',
    salary: '$100k-$130k',
  },
  {
    id: '2',
    title: 'Backend Developer',
    company: 'Tech Inc',
    location: 'Remote',
    description: 'Backend role',
    salary: '$110k-$140k',
  },
];

const mockProps = {
  jobs: mockJobs,
  userId: 'user123',
  onStackEmpty: jest.fn(),
  onError: jest.fn(),
};

describe('SwipeDeck', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the first job card initially', () => {
    render(<SwipeDeck {...mockProps} />);
    
    expect(screen.getByText('Tech Co')).toBeInTheDocument();
    expect(screen.getByText('Frontend Developer')).toBeInTheDocument();
  });

  it('handles right swipe (interest) correctly', async () => {
    render(<SwipeDeck {...mockProps} />);

    // Simulate right swipe
    const card = screen.getByTestId('swipe-card');
    fireEvent.touchStart(card, { touches: [{ clientX: 0, clientY: 0 }] });
    fireEvent.touchMove(card, { touches: [{ clientX: 200, clientY: 0 }] });
    fireEvent.touchEnd(card);

    await waitFor(() => {
      expect(jobAPI.recordJobInterest).toHaveBeenCalledWith('1', 'user123');
    });

    // Should show next card
    expect(screen.getByText('Tech Inc')).toBeInTheDocument();
  });

  it('handles left swipe (dismiss) correctly', async () => {
    render(<SwipeDeck {...mockProps} />);

    // Simulate left swipe
    const card = screen.getByTestId('swipe-card');
    fireEvent.touchStart(card, { touches: [{ clientX: 200, clientY: 0 }] });
    fireEvent.touchMove(card, { touches: [{ clientX: 0, clientY: 0 }] });
    fireEvent.touchEnd(card);

    await waitFor(() => {
      expect(jobAPI.dismissJob).toHaveBeenCalledWith('1', 'user123');
    });

    // Should show next card
    expect(screen.getByText('Tech Inc')).toBeInTheDocument();
  });

  it('calls onStackEmpty when all cards are swiped', async () => {
    render(<SwipeDeck {...mockProps} />);

    // Swipe through all cards
    for (let i = 0; i < mockJobs.length; i++) {
      const card = screen.getByTestId('swipe-card');
      fireEvent.touchStart(card, { touches: [{ clientX: 0, clientY: 0 }] });
      fireEvent.touchMove(card, { touches: [{ clientX: 200, clientY: 0 }] });
      fireEvent.touchEnd(card);

      await waitFor(() => {
        expect(jobAPI.recordJobInterest).toHaveBeenCalledTimes(i + 1);
      });
    }

    expect(mockProps.onStackEmpty).toHaveBeenCalled();
  });

  it('handles API errors correctly', async () => {
    const error = new Error('API Error');
    (jobAPI.recordJobInterest as jest.Mock).mockRejectedValueOnce(error);

    render(<SwipeDeck {...mockProps} />);

    // Simulate right swipe
    const card = screen.getByTestId('swipe-card');
    fireEvent.touchStart(card, { touches: [{ clientX: 0, clientY: 0 }] });
    fireEvent.touchMove(card, { touches: [{ clientX: 200, clientY: 0 }] });
    fireEvent.touchEnd(card);

    await waitFor(() => {
      expect(mockProps.onError).toHaveBeenCalledWith(error);
    });
  });

  it('matches snapshot', () => {
    const { container } = render(<SwipeDeck {...mockProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
