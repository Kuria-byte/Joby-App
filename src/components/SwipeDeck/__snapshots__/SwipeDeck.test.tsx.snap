// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeDeck matches snapshot 1`] = `
<div
  class="swipe-deck-container"
>
  <div
    class="card-container"
    style="transform: translateX(0px) rotate(0deg); opacity: 1; z-index: 2; position: absolute;"
  >
    <div
      class="job-card"
    >
      <div
        class="job-card-header"
      >
        <div
          class="company-image"
        >
          <img
            alt="Tech Co logo"
            src="/icons/company-default.svg"
          />
        </div>
        <div
          class="company-info"
        >
          <h3
            class="company-name"
          >
            Tech Co
          </h3>
          <div
            class="job-meta"
          >
            <span
              class="location"
            >
              <i
                class="fas fa-map-marker-alt"
              />
               
              New York
            </span>
          </div>
        </div>
      </div>
      <div
        class="job-card-body"
      >
        <h2
          class="job-title"
        >
          Frontend Developer
        </h2>
        <div
          class="salary-badge"
        >
          <i
            class="fas fa-money-bill-wave"
          />
           
          $100k-$130k
        </div>
        <p
          class="job-description"
        >
          Frontend role
        </p>
      </div>
      <div
        class="job-card-footer"
      >
        <div
          class="action-buttons"
        >
          <button
            class="btn-more-info"
          >
            <i
              class="fas fa-info-circle"
            />
            More Info
          </button>
          <button
            class="btn-apply"
          >
            <i
              class="fas fa-paper-plane"
            />
            Apply Now
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card-container"
    style="transform: scale(0.95); opacity: 0.8; z-index: 1; position: absolute;"
  >
    <div
      class="job-card"
    >
      <div
        class="job-card-header"
      >
        <div
          class="company-image"
        >
          <img
            alt="Tech Inc logo"
            src="/icons/company-default.svg"
          />
        </div>
        <div
          class="company-info"
        >
          <h3
            class="company-name"
          >
            Tech Inc
          </h3>
          <div
            class="job-meta"
          >
            <span
              class="location"
            >
              <i
                class="fas fa-map-marker-alt"
              />
               
              Remote
            </span>
          </div>
        </div>
      </div>
      <div
        class="job-card-body"
      >
        <h2
          class="job-title"
        >
          Backend Developer
        </h2>
        <div
          class="salary-badge"
        >
          <i
            class="fas fa-money-bill-wave"
          />
           
          $110k-$140k
        </div>
        <p
          class="job-description"
        >
          Backend role
        </p>
      </div>
      <div
        class="job-card-footer"
      >
        <div
          class="action-buttons"
        >
          <button
            class="btn-more-info"
          >
            <i
              class="fas fa-info-circle"
            />
            More Info
          </button>
          <button
            class="btn-apply"
          >
            <i
              class="fas fa-paper-plane"
            />
            Apply Now
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
